lib/IDE/CompletionOverrideLookup.cpp

```diff
@@ -46,8 +46,7 @@ bool CompletionOverrideLookup::addAccessControl(
   //   public class C: B {
   //     public override func foo() {}
   //   }
-  if (Access < AccessLevel::Public &&
-      !isa<ProtocolDecl>(VD->getDeclContext())) {
+  if (Access < AccessLevel::Public && !isa<ProtocolDecl>(CurrentNominal)) {
     for (auto Conformance : CurrentNominal->getAllConformances()) {
       Conformance->getRootConformance()->forEachValueWitness(
           [&](ValueDecl *req, Witness witness) {
```

Comment:
I wonder if we could use `VD->getSatisfiedProtocolRequirements()` here instead

Reply:
Turns out that we don’t need the check at all because the access level of the function inside the class gets elevated to `public` by a Fix-It, so we don’t need to do any fixing ourselves anymore. I added a test case with the scenario described in the comment (which we didn’t have before AFAICT).